### PR TITLE
Aggregated command clear-user-session

### PR DESCRIPTION
**Pull request by YairGlik**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-11626)

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by jbabazadeh — COMMENTED** [review]
great job!

File: Packs/CommonScripts/Scripts/ClearUserSession/ClearUserSession.py [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+                        if "successfully" in human_readable:
+                            success = True
+                            brands_succeeded.append(MS_GRAPH_BRAND)
+                        else:
+                            brands_failed.append(MS_GRAPH_BRAND)
+                            failed_message += f"\nMG User: {human_readable.lstrip('#').strip()}"
+                            demisto.debug(f"Failed to clear sessions for Microsoft Graph user with ID {microsoft_graph_id}. "
+                                          f"Response details: {readable_outputs}")
+
+                if success:
```
Comment (AradCarmi) [inline Packs/CommonScripts/Scripts/ClearUserSession/ClearUserSession.py:386]: Why not just check that the list of the brand succeeded is not empty?

File: Packs/CommonScripts/Scripts/ClearUserSession/ClearUserSession.py [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+        if brand_name == item.get("Source", ""):
+            return item.get("Value", "")
+
+    demisto.debug(
+        f"Skipping user session clearance for user '{user_name}' and brand '{brand_name}' - user name not found."
+    )
+    return ""
+
+
+def clear_user_sessions(command: Command) -> tuple[list[CommandResults], str, Optional[str]]:
```
Comment (AradCarmi) [inline Packs/CommonScripts/Scripts/ClearUserSession/ClearUserSession.py:252]: Please add doc string.

File: Packs/CommonScripts/Scripts/ClearUserSession/ClearUserSession.py [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+    for user in users:
+        username = user.get("Username")
+        id_info = user.get("ID", [])
+        if username and id_info:
+            user_id_mapping[username] = id_info
+
+    return user_id_mapping
+
+
+def get_user_data(command: Command) -> tuple[list[CommandResults], dict]:
```
Comment (AradCarmi) [inline Packs/CommonScripts/Scripts/ClearUserSession/ClearUserSession.py:212]: Please add a doc string.

File: Packs/CommonScripts/Scripts/ClearUserSession/ClearUserSession.py [hunk 4]
```diff
@@ -0,0 +1,421 @@
...
+                )
+            )
+        else:
+            human_readable_list.append(entry.get("HumanReadable") or "")
+    human_readable = "\n".join(human_readable_list)
+    demisto.debug(f"Finished executing command: {command_name}")
+    return entry_context_list, human_readable, errors_command_results
+
+
+def remove_system_user(users_names: list[str]) -> tuple[list, dict]:
```
Comment (AradCarmi) [inline Packs/CommonScripts/Scripts/ClearUserSession/ClearUserSession.py:165]: Please add a doc string.

**Review by AradCarmi — APPROVED** [review]
Great job! 
I approve, please see my notes.

File: Packs/CommonScripts/Scripts/ClearUserSession/ClearUserSession.py [hunk 5]
```diff
@@ -0,0 +1,421 @@
...
+def main():
+    try:
+        args = demisto.args()
+        users_names = argToList(args.get("user_name", ""))
+        verbose = argToBoolean(args.get("verbose", False))
+        brands = argToList(args.get("brands", "")) or DEFAULT_BRANDS
+
+        results_for_verbose: list[CommandResults] = []
+
+        users_names, outputs = remove_system_user(users_names)
```
[the remark this answers is not in the document]
Comment (YairGlik) [reply Packs/CommonScripts/Scripts/ClearUserSession/ClearUserSession.py:313]: I fixed for `users_names`
 Regarding `output` I left The variable does not exist before

File: Packs/CommonScripts/Scripts/ClearUserSession/ClearUserSession.py (hunk 1 above)
Comment (YairGlik) [reply Packs/CommonScripts/Scripts/ClearUserSession/ClearUserSession.py:386]: Right, great idea.

File: Packs/CommonScripts/Scripts/ClearUserSession/ClearUserSession.py [hunk 6]
```diff
@@ -0,0 +1,421 @@
...
+    def __init__(self, name: str, args: dict, brand: Optional[str] = None) -> None:
+        """
+        Initialize a Command object.
+
+        Args:
+            name (str): The name of the command.
+            args (dict): A dictionary containing the command arguments.
+            brand (str, optional): The brand associated with the command. Default is None.
+        """
+        self.brand = brand
```
[the remark this answers is not in the document]
Comment (YairGlik) [reply Packs/CommonScripts/Scripts/ClearUserSession/ClearUserSession.py:24]: why?

File: Packs/CommonScripts/Scripts/ClearUserSession/README.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+## Inputs
+
+---
+
+| **Argument Name** | **Description** |
+| --- | --- |
+| user_name | List of names of the users to retrieve. |
+| brands | Which integrations brands to run the command for. If not provided, the command will run for all available integrations.  For multi-select provide a comma-separated list. For example: "Okta v2,Microsoft Graph User". |
+| verbose | Whether to retrieve human readable entry for every command or only the final result. True means to retrieve human readable entry for every command. False means to human readable only for the final result. |
```
Comment (ShirleyDenkberg) [inline Packs/CommonScripts/Scripts/ClearUserSession/README.md:20]:
```suggestion
| verbose | Whether to retrieve human readable entry for every command or only the final result. True means to retrieve human readable entry for every command. False means to retrieve human readable entry only for the final result. |
```

File: Packs/CommonScripts/Scripts/ClearUserSession/README.md [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+---
+
+| **Path** | **Description** | **Type** |
+| --- | --- | --- |
+| SessionClearingResults.UserName | The username for which the session clearing process was executed. | `string` |
+| SessionClearingResults.Entity | The username for which the session clearing process was executed.| `string`     |
+| SessionClearingResults.Result | The result of the session clearing process for the user (*Success* or *Failed*). | `string` |
+| SessionClearingResults.Brands | The integrations (e.g., *Microsoft Graph*, *Okta v2*) where the session clearing succeeded. | `string` |
+| SessionClearingResults.Message | Additional information or error details if the session clearing process failed. | `string` |
```
Comment (ShirleyDenkberg) [inline Packs/CommonScripts/Scripts/ClearUserSession/README.md:32]: This list of outputs does not match the list of outputs in the YML file nor the path. 
The path in the YML file is SessionClearResults
The path in the README file is SessionCleaingrResults
AND
The YML files lists: SessionClearResults.Source this does not appear in the README.
The README has:  SessionClearingResults.Entity and SessionClearingResults.Brands and these do not appear in the YML file.

**Comment by ShirleyDenkberg** [comment]
@jbabazadeh @AradCarmi Doc review completed.

**Pull request merged**